thunderfish/powerspectrum.py

```diff
@@ -1,13 +1,20 @@
+"""
+
+"""
+
 import numpy as np
-import matplotlib.mlab as ml
+import matplotlib.mlab as mlab
 
-def psd(data, samplerate, fresolution):
+def psd(data, samplerate, fresolution, detrend=mlab.detrend_none,
+    window=mlab.window_hanning, noverlap=None, pad_to=None,
+    sides='default', scale_by_freq=None):
```

Comment:
You should give noverlap also a special meaning, that differs from the one of the mlab.psd function, because you do not know what nfft will be. Appropriate would be for example to specify the overlap as a fraction of nfft. overlap=1 full overlap (makes no sense), overlap = 0 no overlap, overlap = 0.5: overlap of nfft/2.